e2e/jest.test-setup.ts

```diff
@@ -36,37 +34,24 @@ afterEach(async () => {
  */
 beforeAll(async () => {
   await page.setRequestInterception(true);
-  page.on('request', (request) => {
-    const requestUrl = url.parse(request.url(), true);
-    const host = requestUrl.hostname;
-    // to improve page load performance, block network requests unrelated to application.
-    try {
-      if (host === 'www.google-analytics.com'
-           || host === 'accounts.youtube.com'
-           || host === 'static.zdassets.com'
-           || host === 'play.google.com'
-           || request.url().endsWith('content-security-index-report')) {
-        request.abort();
-      } else {
-        request.continue();
-      }
-    } catch (err) {
-      console.error(err);
-    }
+
+  // Log failed requests: 4xx and 5xx status code.
+  page.on('requestfailed', request => {
+    console.error(`❌ Failed request => ${request.method()} ${request.url()} ${request.response().status()}`);
```

Comment:
I'm wondering if this will be too noisy interleaved with the other logs. Have you looked at this library at all? https://github.com/puppeteer/puppeteer/blob/main/docs/api.md#class-tracing

I wonder whether we could just save the trace as an artifact - it should include full information about all network requests made. Sometimes, it may be useful to see successful requests as well in order to debug a test failure.